tests/FakeItEasy.Tests/Configuration/AssertConfigurationTests.cs

```diff
@@ -0,0 +1,73 @@
+namespace FakeItEasy.Tests.Configuration
+{
+    using System.Collections.Generic;
+    using System.Linq.Expressions;
+    using FakeItEasy.Configuration;
+    using FakeItEasy.Tests;
+    using FluentAssertions;
+    using Xunit;
+
+    /// <summary>
+    /// Tests that apply to all implementations of <see cref="IAssertConfiguration"/>.
+    /// </summary>
+    public class AssertConfigurationTests
+    {
+        public interface IFoo
+        {
+            void VoidMethod();
+
+            int WriteableProperty { get;  set; }
+        }
+
+        public static IEnumerable<object[]> AllKindsOfConfigurations()
+        {
+            var unnaturalFake = new Fake<IFoo>();
+            var fake = unnaturalFake.FakedObject;
+            return TestCases.FromObject(
+                A.CallTo(() => fake.VoidMethod()),
+                unnaturalFake.AnyCall(),
+                unnaturalFake.AnyCall().WithNonVoidReturnType(),
+                A.CallToSet(() => fake.WriteableProperty),
+                A.CallToSet(() => fake.WriteableProperty).To(7));
+        }

```

Comment:
Nice! But why use an unnatural fake?

Reply:
It was the call that I found that got me the assertion object I wanted.